Python/Product/PythonTools/PythonTools/Repl/Completion/MarkdownUtil.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+
+namespace Microsoft.PythonTools.Repl.Completion {
+    internal static class MarkdownUtil {
+        // This regex matches markdown code blocks. It extracts the content into two groups:
+        // the language name and the code content.
+        private const string CodeBlockRegex = "^```(.*)[\\S\\s]^([\\S\\s]*?)[\\S\\s]^```";
```

Comment:
i forget if its possible in C# to pre compile these regex strings

Reply:
Yeah it is. I'll change it.